### ISSUE TITLE
XGBoost important parameters to tune

### ISSUE THREAD

**Issue by WatershedXiaolan** [WatershedXiaolan/Xiaolan-s-Gitbook#179]
### Control overfitting by control model complexity
- max_depth
- min_child_weight: minimum sum of instance weight (hessian) needed in a child
- gamma: Minimum loss reduction required to make a further partition on a leaf node of the tree

### Control overfitting by adding randomness 
- subsample: Subsample ratio of the training instances. Setting it to 0.5 means that XGBoost would randomly sample half of the training data prior to growing trees. and this will prevent overfitting.
- colsample_bytree: subsample ratio of columns when constructing each tree

### Faster Training performance 
- tree_method

### Handle imbalanced dataset 
- scale_pos_weight: balance positive and negative weights
